pandas/tseries/resample.py

```diff
@@ -319,9 +319,19 @@ def aggregate(self, arg, *args, **kwargs):
         self._set_binner()
         result, how = self._aggregate(arg, *args, **kwargs)
         if result is None:
-            return self._groupby_and_aggregate(arg,
-                                               *args,
-                                               **kwargs)
+            # It seems there is an ambiguity in having both named args
+            # and variadic in the same function, so we need to distinguish here,
+            # otherwise the first element of args is "eaten" by grouper in
+            # _groupby_and_aggregate.
+            if len(args) == 0:
+                return self._groupby_and_aggregate(arg,
```

Comment:
Sorry, I disappeared for a long time!
Yes, I know it was a hack and I agree it's not a real solution... but I don't know the internals of the code, so I'm not sure I can figure this out...

The problem in my opinion is the definition of the function itself: it seems like you cannot have a function with a named argument with a default value, immediately followed by a variadic part, because the interpreter has no idea where to put the first argument (in the named one or as the first parameter of the variadic?).